broadcast/src/lib.rs

```diff
@@ -1,13 +1,18 @@
 //! Disseminate data over a wide-area network.
 
 use commonware_codec::{Codec, Config};
+use commonware_p2p::Recipients;
+use commonware_utils::Array;
 use futures::channel::oneshot;
 use std::future::Future;
 
 pub mod buffered;
 
 /// Broadcaster is the interface responsible for attempting replication of messages across a network.
 pub trait Broadcaster: Clone + Send + 'static {
+    /// PublicKey is the type of public key used to identify the sender and recipients.
+    type PublicKey: Array;
```

Comment:
I feel that instead of `PublicKey` being the associated type, `Recipients` should be the associated type instead. This makes the trait more flexible. For example, consider "broadcasting" to a certain blockchain, or broadcasting to a URL endpoint. Since this trait is probably not just for P2P broadcast

Reply:
I'm a fan as it removes any relationship to p2p in the trait. Nice.